### ISSUE TITLE
Staff and Students have same Id

### ISSUE THREAD

**Issue by pspaude** [jiscdev/uoj#4]
In using the /stable url for UoJ data we noticed that some Staff and Students have the same STAFF_ID and STUDENT_ID. Since SSP loads everyone as a person... this causes the system to be confused and errors to be generated. If this is not expected in real-world usage, the data should be fixed to have unique Staff and Student Ids.

If this is expected in real-world usage, let us know, as we'll have to look at options for supporting that functionality.

**Comment by amason** [comment]
@michaelwebbjisc

**Comment by michaelwebbjisc** [comment]
yes, this is a valid real-world scenario - staff and student ids may clash.

**Comment by amason** [comment]
Hi @michaelwebbjisc  Were you able to find any additional information around the likelihood of this happening based on Gloucestershire and Cardiff's data? Let us know because today, SSP can't handle this and the system will error out with a 500 error when we run into this scenario. We're going to need to understand more around the following:

- How likely is this scenario to happen in the real world?

- How do the SIS systems handle this today without getting confused? Is there an additional unique ID that comes into play from a mapping perspective?

- Should SSP incorporate a unique ID coming over in the UDD data to avoid this situation?

**Comment by amason** [comment]
@michaelwebbjisc 
Michael/Rob,

In regard to the issue we've been discussing where we were seeing Staff and Students having the same STAFF_ID and STUDENT_ID with University of Jisc data (https://github.com/jiscdev/uoj/issues/4), we decided to run a query against the LWR for all 4 pilot institutions to see if this is a reality based on our 4 pilot institutions we will be working with. We ran a query for Gloucestershire' s data to see if there are duplicate/similar IDs. The results returned the following:

"Here are the characteristics for the UoG student/staff ID's.
in historical data, student/staff ID is 33 char alphanumeric/hex
in live data, student/staff ID is 7 char numeric
Findings:
NO duplicate historical student/staff ID's
NO duplicate live student/staff ID's"


For Aber, Cardiff and Greenwich, these were our findings:
"As yet, there is NO staff UDD data for CardMet, Aber or Greenwich using live or historical credentials. Unable to determine duplicates."

When we have the full sets of data for the 3 institutions, I want to run the queries again at each institutional level but based on these findings with Gloucestershire, I feel we are safe to proceed as normal without making any changes in SSP or in the UDD. Would you guys agree?

I do think it's important for us to write a SPIKE JIRA for the backlog so that if this situation ever happens in Pilot/Prod, we have the history and we can research for the best solution at that time.

Does this sounds like a good plan to you all? Let me know.
